tests/InstrEngineTests/InstrEngineTests/ProfilerHelpers.cs

```diff
@@ -126,66 +186,99 @@ public static string[] SplitXmlDocuments(string content)
+                        }
                     }
                 }
-                outputStr = strBuilder.ToString();
-            }
-
-            string[] baselineDocs = SplitXmlDocuments(baselineStr);
-            string[] outputDocs = SplitXmlDocuments(outputStr);
+                else
+                {
+                    // Use XML comparison
+                    string[] baselineDocs = SplitXmlDocuments(baselineStr);
+                    string[] outputDocs = SplitXmlDocuments(outputStr);
 
-            Assert.AreEqual(baselineDocs.Length, outputDocs.Length);
-            for (int docIdx = 0; docIdx < baselineDocs.Length; docIdx++)
-            {
+                    Assert.AreEqual(baselineDocs.Length, outputDocs.Length);
+                    for (int docIdx = 0; docIdx < baselineDocs.Length; docIdx++)
+                    {
 
-                string baselineXmlDocStr = baselineDocs[docIdx];
-                string outputXmlDocStr = outputDocs[docIdx];
+                        string baselineXmlDocStr = baselineDocs[docIdx];
+                        string outputXmlDocStr = outputDocs[docIdx];
 
-                XmlDocument baselineDocument = new XmlDocument();
-                baselineDocument.LoadXml(baselineXmlDocStr);
+                        XmlDocument baselineDocument = new XmlDocument();
+                        baselineDocument.LoadXml(baselineXmlDocStr);
 
-                XmlDocument outputDocument = new XmlDocument();
-                outputDocument.LoadXml(outputXmlDocStr);
+                        XmlDocument outputDocument = new XmlDocument();
+                        outputDocument.LoadXml(outputXmlDocStr);
 
-                Assert.AreEqual(baselineDocument.ChildNodes.Count, outputDocument.ChildNodes.Count);
+                        Assert.AreEqual(baselineDocument.ChildNodes.Count, outputDocument.ChildNodes.Count);
 
-                for (int i = 0; i < baselineDocument.ChildNodes.Count; i++)
-                {
-                    XmlNode currBaselineNode = baselineDocument.ChildNodes[i];
-                    XmlNode currOutputNode = outputDocument.ChildNodes[i];
+                        for (int i = 0; i < baselineDocument.ChildNodes.Count; i++)
+                        {
+                            XmlNode currBaselineNode = baselineDocument.ChildNodes[i];
+                            XmlNode currOutputNode = outputDocument.ChildNodes[i];
 
-                    DiffResultToBaselineNode(currBaselineNode, currOutputNode);
+                            DiffResultToBaselineNode(currBaselineNode, currOutputNode);
+                        }
+                    }
                 }
             }
+            catch (AssertFailedException)
+            {
+                Console.WriteLine($"Baseline FilePath: {baselinePath}");
+                Console.WriteLine();
+                Console.WriteLine($"Output FilePath: {outputPath}");
+                throw;
+            }
```

Comment:
I've found it annoying to figure out which folder in TestResult\ corresponded to the failure. Luckily VSTest outputs stdout so I'm dumping these filepaths on failures